custom_model_runner/datarobot_drum/drum/adapters/model_adapters/python_model_adapter.py

```diff
@@ -107,43 +104,35 @@ def __init__(self, model_dir, target_type=None):
         # New custom task class and instance loaded from custom.py
         self._custom_task_class = None
         self._custom_task_class_instance = None
-        self._guard_moderation_hooks = None
-        self._guard_pipeline = None
+        self._moderation_pipeline = None
+        self._moderation_score_fn = None
+        self._moderation_chat_fn = None
 
         if target_type in (TargetType.TEXT_GENERATION, TargetType.VECTOR_DATABASE):
             self._target_name = os.environ.get("TARGET_NAME")
             if not self._target_name:
                 raise ValueError(
                     "Unexpected empty target name for text generation or vector database target."
                 )
-            if target_type == TargetType.TEXT_GENERATION:
-                self._load_guard_hooks_for_drum()
+            self._load_moderation_hooks()
         else:
             self._target_name = None
 
-    def _load_guard_hooks_for_drum(self):
+    def _load_moderation_hooks(self):
         try:
-            guard_module = __import__(GUARD_HOOK_MODULE, fromlist=[MODERATIONS_LIBRARY_PACKAGE])
+            mod_module = __import__(MODERATIONS_HOOK_MODULE, fromlist=[MODERATIONS_LIBRARY_PACKAGE])
             self._logger.info(
-                f"Detected {guard_module.__name__} in {guard_module.__file__}.. "
-                f"trying to load hooks"
+                f"Detected {mod_module.__name__} in {mod_module.__file__}.. trying to load hooks"
             )
-            self._guard_moderation_hooks = {
-                GUARD_INIT_HOOK_NAME: getattr(guard_module, GUARD_INIT_HOOK_NAME, None),
-                GUARD_SCORE_WRAPPER_NAME: getattr(guard_module, GUARD_SCORE_WRAPPER_NAME, None),
-                GUARD_CHAT_WRAPPER_NAME: getattr(guard_module, GUARD_CHAT_WRAPPER_NAME, None),
-            }
-            if self._guard_moderation_hooks[GUARD_INIT_HOOK_NAME] and (
-                self._guard_moderation_hooks[GUARD_SCORE_WRAPPER_NAME]
-                or self._guard_moderation_hooks[GUARD_CHAT_WRAPPER_NAME]
-            ):
-                self._guard_pipeline = self._guard_moderation_hooks[GUARD_INIT_HOOK_NAME]()
-            else:
-                self._logger.debug("No guards defined")
+            self._moderation_score_fn = mod_module.get_moderations_fn(self._target_type, "score")
```

Comment:
"score" -> CustomHooks.SCORE
"chat" -> CustomHooks.CHAT

Reply:
I will also call them `moderation_*_hook` instead of `fn` - they are really hooks into moderation library

Reply:
You also need to check existence of `create_pipeline`.  It is possible that moderation library is not updated and will fail.  Ensures backward compatibility